lib/Sema/CSSimplify.cpp

```diff
@@ -1853,12 +1853,13 @@ ConstraintSystem::matchTypesBindTypeVar(
 
   // We do not allow keypaths to go through AnyObject. Let's create a fix
   // so this can be diagnosed later.
-  auto loc = getConstraintLocator(locator);
-  if (loc->isKeyPathRoot() && type->isAnyObject()) {
-    auto *fix = AllowAnyObjectKeyPathRoot::create(*this, loc);
+  if (auto loc = getConstraintLocator(locator)) {
+    if (loc->isKeyPathRoot() && type->isAnyObject()) {
```

Comment:
As a suggestion, you might be better off just doing something like:

```c++
if (auto elt = locator.last()) {
  if (elt->getKind() == ConstraintLocator::KeyPathRoot) {
     ...
  }
}
```

Instead of materializing the whole locator every time.

Reply:
Actually wait a minute! You don't want locator of the constraint here, instead you want locator of the type variable  - `typeVar->getImpl().getLocator()`.

Reply:
You're right! I didn't realise the locator passed to this function doesn't actually point to the type variable. Fixed now!